lib/FieldType/Mapper/SelectionFormMapper.php

```diff
@@ -33,32 +36,71 @@ class SelectionFormMapper implements FieldDefinitionFormMapperInterface, FieldVa
                 'required' => false,
                 'property_path' => 'fieldSettings[isMultiple]',
                 'label' => 'field_definition.ezselection.is_multiple',
                 'disabled' => $isTranslation,
-            ])
-            ->add('options', CollectionType::class, [
-                'entry_type' => TextType::class,
-                'entry_options' => ['required' => false],
-                'allow_add' => true,
-                'allow_delete' => true,
-                'delete_empty' => false,
-                'prototype' => true,
-                'prototype_name' => '__number__',
-                'required' => false,
-                'property_path' => 'fieldSettings[options]',
-                'label' => 'field_definition.ezselection.options',
-                'disabled' => $isTranslation,
             ]);
+
+        $formBuilder = $fieldDefinitionForm->getConfig()->getFormFactory()->createBuilder();
+
+        $optionField = $formBuilder->create($optionFieldName, CollectionType::class, [
+            'entry_type' => TextType::class,
+            'entry_options' => ['required' => false],
+            'allow_add' => true,
+            'allow_delete' => true,
+            'delete_empty' => false,
+            'prototype' => true,
+            'prototype_name' => '__number__',
+            'required' => false,
+            'property_path' => 'fieldSettings[' . $optionFieldName . ']',
+            'label' => 'field_definition.ezselection.options',
+        ]);
+
+        if ($isMultilingual) {
+            $dataListener = new SelectionMultilingualOptionsDataListener($languageCode);
+            $dataTransformer = new MultilingualSelectionTransformer($languageCode, $data);
+
+            $optionField
+                ->addEventListener(
+                    FormEvents::PRE_SET_DATA,
+                    [$dataListener, 'setLanguageOptions'],
+                    10
+                )
+                ->addModelTransformer(
+                    $dataTransformer
+                );
+        }
+
+        $fieldDefinitionForm->add(
+            $optionField
+                ->setAutoInitialize(false)
+                ->getForm()
+        );
     }
 
     public function mapFieldValueForm(FormInterface $fieldForm, FieldData $data)
     {
         $fieldDefinition = $data->fieldDefinition;
         $formConfig = $fieldForm->getConfig();
+        $languageCode = $fieldForm->getConfig()->getOption('languageCode');
+
+        $choices = $fieldDefinition->fieldSettings['options'];
+
+        if (!empty($fieldDefinition->fieldSettings['multilingualOptions'][$fieldDefinition->mainLanguageCode])) {
+            $choices = $fieldDefinition->fieldSettings['multilingualOptions'][$fieldDefinition->mainLanguageCode];
+        }
+
+        if (!empty($fieldDefinition->fieldSettings['multilingualOptions'][$languageCode])) {
+            $choices = $fieldDefinition->fieldSettings['multilingualOptions'][$languageCode];
```

Comment:
These two if blocks look like a case for if-else, probably with reversed order, to reduce number of comparisons.

Reply:
changd